applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py

```diff
@@ -29,45 +29,85 @@
+        
+        self.WriteDesignSurface = False
+        self.WriteOptimizationModelPart = False
+
+        if OutputMode == "WriteDesignSurface":
+            self.WriteDesignSurface = True
+        elif OutputMode == "WriteOptimizationModelPart":
+            if self.OptimizationModelPart.NumberOfElements() == 0:
+                raise NameError("Output of optimization model part in Gid-format requires definition of elements. No elements are given in current mdpa! You may change the design output mode.")              
+            self.WriteOptimizationModelPart = True
+        else:
+            raise NameError("The following design output mode is not defined within a GiD output (name may be misspelled): " + OutputMode)              
+
+    # --------------------------------------------------------------------------
+    def __CreateGiDIO( self ):
+        self.__ModifySettingsToMatchDefaultGiDOutputProcess()        
+
+        GidConfig = self.OutputSettings["output_format"]["gid_configuration"]
+        ResultsDirectory = self.OutputSettings["output_directory"].GetString()
+        DesignHistoryFilename = self.OutputSettings["design_history_filename"].GetString()
+        DesignHistoryFilenameWithPath =  ResultsDirectory+"/"+DesignHistoryFilename
+
+        if self.WriteDesignSurface:     
+            self.GidIO = GiDOutputProcess(self.DesignSurface, DesignHistoryFilenameWithPath, GidConfig)
+        elif self.WriteOptimizationModelPart:
+            self.GidIO = GiDOutputProcess(self.OptimizationModelPart, DesignHistoryFilenameWithPath, GidConfig)
+
+    # --------------------------------------------------------------------------
+    def __ModifySettingsToMatchDefaultGiDOutputProcess( self ):      
+        self.__AddNodalResultsToGidConfiguration()
+        self.__SetConditionsFlagAccordingOutputMode()
+
+    # --------------------------------------------------------------------------
+    def __AddNodalResultsToGidConfiguration( self ):
+        NodalResults = self.OutputSettings["nodal_results"]        
+        self.OutputSettings["output_format"]["gid_configuration"]["result_file_configuration"].AddValue("nodal_results", NodalResults)
+        
+    # --------------------------------------------------------------------------
+    def __SetConditionsFlagAccordingOutputMode( self ):  
+        GidConfig = self.OutputSettings["output_format"]["gid_configuration"]
+        
+        if not GidConfig["result_file_configuration"]["gidpost_flags"].Has("WriteConditionsFlag"):           
+            GidConfig["result_file_configuration"]["gidpost_flags"].AddEmptyValue("WriteConditionsFlag")
+
+        if self.WriteDesignSurface:     
+            GidConfig["result_file_configuration"]["gidpost_flags"]["WriteConditionsFlag"].SetString("WriteConditions")
+        elif self.WriteOptimizationModelPart:
+            GidConfig["result_file_configuration"]["gidpost_flags"]["WriteConditionsFlag"].SetString("WriteElementsOnly")       
```

Comment:
WriteConditionsFlag seems to be the wrong name? as it can contain the WriteELEMENTSOnly string?

Reply:
Forget what i said, if this is a default GIDIO setting